spectrocloud/resource_cluster_profile.go

```diff
@@ -64,17 +64,22 @@ func resourceClusterProfile() *schema.Resource {
 				Optional: true,
 			},
 			"cloud": {
-				Type:     schema.TypeString,
-				Default:  "all",
-				Optional: true,
-				ForceNew: true,
+				Type:         schema.TypeString,
+				Default:      "all",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"all", "aws", "azure", "gcp", "vsphere", "openstack", "maas", "nested", "baremetal", "eks", "aks", "edge", "edge-native", "libvirt", "tencent", "tke", "coxedge", "generic", "gke"}, false),
+				ForceNew:     true,
+				Description: "A cloud provider type supported in Palette, Supported cloud types are - `all, aws, azure, gcp, vsphere, openstack, maas, nested, baremetal, eks, aks, edge, edge-native, libvirt, tencent, tke, coxedge, generic and gke`," +
+					"If a cloud value is `all` then type must be `add-on`. Otherwise it should be a corresponding cloud type.",
```

Comment:
```suggestion
				Description: "Specify the infrastructure provider the cluster profile is for. Only Palette supported infrastructure providers can be used. The  supported cloud types are - `all, aws, azure, gcp, vsphere, openstack, maas, nested, baremetal, eks, aks, edge, edge-native, libvirt, tencent, tke, coxedge, generic, and gke`," +
					"If the value is set to `all`, then the type must be set to `add-on`. Otherwise, the cluster profile may be incompatible with other providers.",
```
I added some clarification because the term `cloud` may lead to some thinking only cloud providers can be used.